telluric/vrt.py

```diff
@@ -136,4 +152,87 @@ def boundless_vrt_doc(
         dst_window = Window(xoff, yoff, xsize, ysize)
         vrt.add_band_simplesource(mask_band, 'mask,1', 'Byte', False, src_dataset.name,
                                   width, height, block_shape[1], block_shape[0], src_window, dst_window)
-    return vrt.tostring()
+    return vrt
+
+
+def band_name_to_color_interpretation(band_name):
+    if band_name.lower() in ['red', 'green', 'blue']:
+        return band_name
+    if band_name.lower().endswith('_enhanced'):
+        return band_name_to_color_interpretation(band_name[:-len('_enhanced')])
+    else:
+        return 'grey'
+
+
+def raster_list_vrt(rasters, relative_to_vrt=True):
+    """Make a VRT XML document from a list of GeoRaster2 objects.
+    Parameters
+    ----------
+    rasters : list
+        The list of GeoRasters.
+    relative_to_vrt : bool, optional
+        If True the bands simple source url will be related to the VRT file
+    Returns
+    -------
+    bytes
+        An ascii-encoded string (an ElementTree detail)
+    """
+
+    from telluric import FeatureCollection
+    fc = FeatureCollection.from_georasters(rasters)
+    return raster_collection_vrt(fc, relative_to_vrt)
+
+
+def raster_collection_vrt(fc, relative_to_vrt=True):
+    """Make a VRT XML document from a feature collection of GeoRaster2 objects.
+    Parameters
+    ----------
+    rasters : FeatureCollection
+        The FeatureCollection of GeoRasters.
+    relative_to_vrt : bool, optional
+        If True the bands simple source url will be related to the VRT file
+    Returns
+    -------
+    bytes
+        An ascii-encoded string (an ElementTree detail)
+    """
+
+    def max_resolution():
+        max_affine = max(fc, key=lambda f: f.raster.resolution()).raster.affine
+        return abs(max_affine.a), abs(max_affine.e)
+
+    from telluric import rasterization
+    assert all(fc.crs == f.crs for f in fc), "all rasters should have the same CRS"
+
+    rasters = (f.raster for f in fc)
+    bounds = fc.convex_hull.get_bounds(fc.crs)
+    resolution = max_resolution()
+    width, height, affine = rasterization.raster_data(bounds, resolution)
+
+    bands = {}  # type: Dict[str, tuple]
+    vrt = BaseVRT(width, height, fc.crs, affine)
+
+    last_band_idx = 0
+    for raster in rasters:
```

Comment:
don't we need to add the mask band where applicable?

Reply:
the VRT restrict the use of a single mask band which is something I cannot implement when dealing with multiple rasters

Reply:
when you run `gdalbuildvrt` on files with mask, it creates a single mask band consist of all rasters when we have a single band

We have a problem with multibands, it mean that at the moment we will support only no-data masking